### PR TITLE
feat: update device

### LINKED ISSUE #48: Use case for updating a device

### PR DESCRIPTION
**Pull request by oriolagobat**
Closes #48

### CODE REVIEW

**Review by FerranAD — APPROVED** [review]

File: src/test/kotlin/device/application/DeviceCreatorTest.kt [hunk 1]
```diff
@@ -27,10 +27,10 @@ class DeviceCreatorTest {
 
     @Test
     fun `If trying to create a device with invalid id, InvalidUUIDException is thrown`() {
-        val invalidId = DeviceIdentifierMother.getInvalidIdentifierPrimitive()
+        val invalidIdPrimitive = IdentifierMother.getInvalidIdentifierPrimitive()
 
         assertThrows<InvalidUUIDException> {
-            DeviceIdentifier(invalidId)
+            Identifier(invalidIdPrimitive)
```
Comment (Pablito2020) [inline src/test/kotlin/device/application/DeviceCreatorTest.kt:33]: This should test the use case, not the identifier implementation

File: src/test/kotlin/shared/mothers/IdentifierMother.kt [hunk 2]
```diff
@@ -13,4 +13,5 @@ object IdentifierMother {
         return Identifier(uuid)
     }
     const val invalidPrimitiveIdentifier = "InvalidUUID-12345"
+    fun getInvalidIdentifierPrimitive() = invalidPrimitiveIdentifier
```
Comment (Pablito2020) [inline src/test/kotlin/shared/mothers/IdentifierMother.kt:16]: Not needed, you can access it via the constant (invalidPrimitiveIdentifier), a funtion isn't really needed

File: src/test/kotlin/device/mothers/DeviceMother.kt [hunk 3]
```diff
@@ -8,6 +8,11 @@ object DeviceMother {
         id = IdentifierMother.getValidIdentifier(),
         name = DeviceNameMother.getValidDeviceName(),
     )
-    fun getIdPrimitiveFrom(device: Device) = device.id.id
-    fun getNamePrimitiveFrom(device: Device) = device.name.name
+
+    fun getDeviceWithDifferentNameFrom(device: Device) = device.copy(
```
Comment (Pablito2020) [inline src/test/kotlin/device/mothers/DeviceMother.kt:12]: The copy isn't really needed if you change all the fields (like you are doing here). If you change all the fields a new instatiaton is less confusing and verbose

**Review by Pablito2020 — CHANGES_REQUESTED** [review]
The next time, is MANDATORY to do only one refactor/feature for pull request, if it doesn't follow the standards accorded on #43 , it will be closed immediately.

And please, be aware to test the use case, not the value object implementations

**Review by Pablito2020 — APPROVED** [review]

**Pull request merged**